contracts/strategies/PenguinIglooStrategyV2.sol

```diff
@@ -0,0 +1,247 @@
+        require(
+            DexLibrary.checkSwapPairCompatibility(IPair(_swapPairWAVAXPEFI), address(WAVAX), address(_rewardToken)),
+            "_swapPairWAVAXPEFI is not a WAVAX-Pefi pair"
+        );
+        require(
+            _swapPairToken0 == address(0)
+            || DexLibrary.checkSwapPairCompatibility(IPair(_swapPairToken0), address(WAVAX), IPair(address(depositToken)).token0()),
+            "_swapPairToken0 is not a WAVAX+deposit token0"
+        );
+        require(
+            _swapPairToken1 == address(0)
+            || DexLibrary.checkSwapPairCompatibility(IPair(_swapPairToken1), address(WAVAX), IPair(address(depositToken)).token1()),
+            "_swapPairToken0 is not a WAVAX+deposit token1"
+        );
+        // converts Pefi to WAVAX
+        swapPairWAVAXPEFI = IPair(_swapPairWAVAXPEFI);
+        // converts WAVAX to pair token0
+        swapPairToken0 = IPair(_swapPairToken0);
+        // converts WAVAX to pair token1
+        swapPairToken1 = IPair(_swapPairToken1);
+    }
+
+  /**
+   * @notice Deposit tokens to receive receipt tokens
+   * @param amount Amount of tokens to deposit
+   */
+  function deposit(uint amount) external override {
+    _deposit(msg.sender, amount);
+  }
+
+  /**
+   * @notice Deposit using Permit
+   * @param amount Amount of tokens to deposit
+   * @param deadline The time at which to expire the signature
+   * @param v The recovery byte of the signature
+   * @param r Half of the ECDSA signature pair
+   * @param s Half of the ECDSA signature pair
+   */
+  function depositWithPermit(uint amount, uint deadline, uint8 v, bytes32 r, bytes32 s) external override {
+    depositToken.permit(msg.sender, address(this), amount, deadline, v, r, s);
+    _deposit(msg.sender, amount);
+  }
+
+  function depositFor(address account, uint amount) external override {
+      _deposit(account, amount);
+  }
+
+  function _deposit(address account, uint amount) internal {
+    require(DEPOSITS_ENABLED == true, "PenguinIglooStrategyV2::_deposit");
+    if (MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST > 0) {
+        uint unclaimedRewards = checkReward();
+        if (unclaimedRewards > MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST) {
+            _reinvest(unclaimedRewards);
+        }
+    }
+        require(depositToken.transferFrom(msg.sender, address(this), amount), "PenguinIglooStrategyV2::transfer failed");
+    _stakeDepositTokens(amount);
+    _mint(account, getSharesForDepositTokens(amount));
+    emit Deposit(account, amount);
+  }
+
+  function withdraw(uint amount) external override {
+    uint depositTokenAmount = getDepositTokensForShares(amount);
+    if (depositTokenAmount > 0) {
+      _withdrawDepositTokens(depositTokenAmount);
+      (,,,,,, uint withdrawFeeBP,,) = stakingContract.poolInfo(PID);
+      uint withdrawFee = depositTokenAmount.mul(withdrawFeeBP).div(BIPS_DIVISOR);
```

Comment:
not sure if this it's a good idea to use BIPS_DIVISOR, are we sure they will always match PEFI bips?

Reply:
Yep! Their MAX_BIPS which is equivalent to our BIPS_DIVISOR is an `uint256 internal constant` in their IglooMasterChef  and is set to `10000` so i think we are good with it.

Reply:
I'd prefer decoupling as we might change our own BIPS_DIVISOR as it's getting close to a moment where there's not enough digits

Reply:
Updated the code to take it in constructor and included an update method for the owner to manage the variable.